tests/test_task_collection_pypi.py

```diff
@@ -158,6 +158,50 @@ async def test_pip_install(tmp_path):
     assert PACKAGE in location.as_posix()
 
 
+async def test_pip_install_pinned(tmp_path):
+    """ """
+
+    PACKAGE = "devtools"
+    VERSION = "0.8.0"
+    EXTRA = "pygments"
+    PIN_VERSION = "2.0"
+    PIN = {EXTRA: PIN_VERSION}
+
+    for pin in [None, PIN]:
```

Comment:
At first it was unclear to me why to include `None`.

Let's add a comment, like "We also include `None`, to make sure that the test with `PIN` is not passing for an arbitrary coincidence (e.g because `devtools[pygments]==0.8.0` already installed `pygments=2.0`)"